GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -163,34 +164,49 @@ public async Task ReloadAsync()
                 treeMain.BeginUpdate();
                 _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = true);
                 var tasks = _rootNodes.Select(r => r.ReloadAsync(token)).ToArray();
-                await Task.WhenAll(tasks).ConfigureAwait(false);
+                await Task.WhenAll(tasks).ConfigureAwait(true);
```

Comment:
Changed this to ConfigureAwait(true) here so that we come back on the UI thread when done. This avoids having to call SwitchToMainThreadAsync just below, which I replaced with asserting we're on the UI thread.

Reply:
Do you think so? I mean, that's what ConfigureAwait(true) means, and the assertion just below documents the expectation. If you still want me to add a comment, I can.

Reply:
I'd do it for a few reasons.
1. Top guys like yourself who deal with async stuff everyday would probably understand the nature of the change without a comment. But for other contributors who don't have a good understanding of threading or don't deal with it everyday the change may look puzzling
2. Going through the blame one wouldn't necessarily locate this discussion and read the reasons for the change
3. A comment in itself may help others learn

Reply:
Is there any reason to start executing the finally before all tasks are done? This seems like the normal way to implement this, the previous way was a solution that happened to work. If so, no comment should be needed.

Reply:
You guys decide whether to add a comment. Just wanted to add that I didn't know anything about async programming in C# a couple of months ago, but had to learn it to work on GE, and knowing how ConfigureAwait works is part of that. But if you think it will help get others up to speed, I'll add a comment.

Reply:
> the previous way was a solution that happened to work.

There are many ways achieving the desired. We're already arguing about this longer than it would take to add the comment and merge :)

Reply:
`ConfigureAwait(true)` is a no-op. In general I think we should be avoiding use of ConfigureAwait everywhere, and instead use JTF to be explicit about which thread we want to be on.

https://github.com/Microsoft/vs-threading/blob/master/doc/cookbook_vs.md#should-i-await-a-task-with-configureawaitfalse

Reply:
I captured some info here: https://github.com/gitextensions/gitextensions/wiki/Async-guidelines
Feel free to add more